File: src/main/java/Day01.java
import java.util.HashSet;
import java.util.List;

@SuppressWarnings("unused")
public class Day01 implements IAocTask {

    private HashSet<Integer> frequencies = new HashSet<>();

    @Override
    public String getFileName() {
        return "aoc2018/input_01.txt";
    }

    @Override
    public void solvePartOne(List<String> lines) {

        int result = 0;

        for (String line : lines) {
            result += Integer.parseInt(line.trim());
        }

        System.out.println(result);
    }

    @Override
    public void solvePartTwo(List<String> lines) {
        int currentFrequency = 0;
        frequencies.add(currentFrequency);

        for (int i = 0; i < lines.size(); i = (i + 1) % lines.size()) {
            currentFrequency += Integer.parseInt(lines.get(i).trim());
            if (frequencies.contains(currentFrequency)) {
                System.out.println(currentFrequency);
                break;
            }
            frequencies.add(currentFrequency);
        }
    }
}

File: src/main/kotlin/year2024/Day01.kt
package year2024

import aoc.IAocTaskKt
import kotlin.math.abs

class Day01 : IAocTaskKt {
    override fun getFileName(): String =  "aoc2024/input_01.txt"

    override fun solvePartOne(lines: List<String>) {
        val a = lines.map { line -> line.split(" ").first().toInt() }.sorted()
        val b = lines.map { line -> line.split(" ").last().toInt() }.sorted()
        val result = a.zip(b) { x, y -> abs(x - y) }.sum()
        println(result)
    }

    override fun solvePartTwo(lines: List<String>) {
        val a = lines.map { line -> line.split(" ").first().toInt() }.sorted()
        val b = lines.map { line -> line.split(" ").last().toInt() }.sorted()

        var occurrences = 0
        for (num in a) {
            occurrences += num * b.count { it == num }
        }
        println(occurrences)
    }
}